src/common/transformations/src/transformations/common_optimizations/merge_similar_branches.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/merge_similar_branches.hpp"
+
+#include <algorithm>
+#include <queue>
+#include <unordered_set>
+
+#include "openvino/op/util/op_types.hpp"
+
+using namespace std;
+
+namespace ov {
+namespace pass {
+
+namespace {
+
+bool are_equal_constants(const Node* const l, const Node* const r) {
```

Comment:
I believe we have this method in the codebase, is it posible to re-use it?

Reply:
Yes. I moved it to common part `transformations/utils/utils.hpp`